sim/core/spell_result.go

```diff
@@ -292,16 +283,35 @@ func (spell *Spell) calcDamageInternal(sim *Simulation, target *Unit, baseDamage
 }
 func (spell *Spell) CalcDamage(sim *Simulation, target *Unit, baseDamage float64, outcomeApplier OutcomeApplier) *SpellResult {
 	attackerMultiplier := spell.AttackerDamageMultiplier(spell.Unit.AttackTables[target.UnitIndex][spell.CastType])
+	if spell.BonusCoefficient > 0 {
+		baseDamage += spell.BonusCoefficient * spell.BonusDamage()
+	}
 	return spell.calcDamageInternal(sim, target, baseDamage, attackerMultiplier, false, outcomeApplier)
 }
 func (spell *Spell) CalcPeriodicDamage(sim *Simulation, target *Unit, baseDamage float64, outcomeApplier OutcomeApplier) *SpellResult {
 	attackerMultiplier := spell.AttackerDamageMultiplier(spell.Unit.AttackTables[target.UnitIndex][spell.CastType])
+	if spell.Dot(target).BonusCoefficient > 0 {
+		baseDamage += spell.Dot(target).BonusCoefficient * spell.BonusDamage()
+	}
 	return spell.calcDamageInternal(sim, target, baseDamage, attackerMultiplier, true, outcomeApplier)
 }
 func (dot *Dot) CalcSnapshotDamage(sim *Simulation, target *Unit, outcomeApplier OutcomeApplier) *SpellResult {
 	return dot.Spell.calcDamageInternal(sim, target, dot.SnapshotBaseDamage, dot.SnapshotAttackerMultiplier, true, outcomeApplier)
 }
 
+func (dot *Dot) Snapshot(target *Unit, baseDamage float64, isRollover bool) {
+	// Rollovers in SoD don't seem to update anything
+	if !isRollover {
```

Comment:
nit: invert for early out.

Reply:
And once again: I really like this helper, but the "isRollover" support should just be removed.